policies/policy_service.go

```diff
@@ -258,6 +264,17 @@ func (s policiesService) EditDataset(ctx context.Context, token string, ds Datas
 	if err != nil {
 		return Dataset{}, err
 	}
+
+	errValidatePolicy := s.validateDatasetPolicy(ctx, datasetEdited.MFOwnerID, datasetEdited.PolicyID)
+	errValidateAGroup := s.validateDatasetAgentGroup(ctx, datasetEdited.MFOwnerID, datasetEdited.AgentGroupID)
+
+	if errValidatePolicy == nil && errValidateAGroup == nil{
+		err = s.repo.ActivateDatasetByID(ctx, datasetEdited.ID, datasetEdited.MFOwnerID)
+		if err != nil{
+			return Dataset{}, err
+		}
+	}
```

Comment:
@weyrick yeah, you're right, but the validation process on these two resources is meant to make sure the dataset has been invalidated because of a deleted sink and not by anything else, so then by editing and adding a valid sink, the dataset can become valid again. Otherwise, we can still edit, change name, tags, and even linked sinks, but it would never become valid again. We can print something in logs to be visible but the service should not return at this point if it gets something from these errors just check if it can be valid or not. What do you think?

Reply:
ok I guess what we need is to determine the best way for the product to let them know the reason a thing is "invalid" in the system. that can be separate work.